WordPress/gutenberg
# Heading Block: Don't rely on the experimental selector anymore

## What?
Before we had a class name on the heading block, the only way to target it was using this selector `h1,h2,h3,h4,h5,h6`. Now we have a class name on the block (https://github.com/WordPress/gutenberg/pull/42122), we can remove this __experimentalSelector and revert to the standard selector (`.wp-block-heading`).

## Why?
This gives theme.json more control over what it targets - now heading block settings will only apply to heading blocks, not all heading elements.

## How?
Simply remove the __experimentalSelector property.

## Testing Instructions
0. Switch to a block theme
1. Add this to your theme.json file:
```
			"core/heading": {
				"color": {
					"background": "red"
				}
			},
```
3. Check that the heading block has a red background, but other block that use H* elements do not (for example Post Title).
 
### Testing Instructions for Keyboard


## Screenshots or screencast 
<img width="867" alt="Screenshot 2022-12-02 at 17 02 49" src="https://user-images.githubusercontent.com/275961/205345913-8bd094d4-f5c1-4467-8552-d5feda9a48ac.png">

@WordPress/block-themers
